File: Backend/EComCore.Domain/Entities/Address.cs
namespace EComCore.Domain.Entities;

public class Address
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string? AddressLine1 { get; set; }
    public string? AddressLine2 { get; set; }
    public string City { get; set; }
    public string PostalCode { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public bool IsDeleted { get; set; }
}

File: client/src/types/address.ts
export interface Address {
  id: string;
  name: string;
  addressLine1: string;
  addressLine2: string;
  city: string;
  postalCode: string;
  isDeleted: string;
}

export interface CreateAddress {
  name: string;
  addressLine1: string;
  addressLine2: string;
  city: string;
  postalCode: string;
  userId: number;
}
export interface UpdateAddress {
  name: string;
  addressLine1: string;
  addressLine2: string;
  city: string;
  postalCode: string;
}
